From: Stefan Kerkmann <s.kerkmann@pengutronix.de>
Date: Wed, 22 Jan 2025 15:58:29 +0100
Subject: [PATCH] i.mx8: ocotp: add GPx_LOCK defines

The GPx_LOCK registers changed position in the fuse map from the i.MX6
series and derivates to the i.MX8 series with i.MX8MP being different
again.

The defines have been derived from:

IMX8MNRM Rev. 2 (i.MX8MN)
IMX8MPRM Rev. 3 (i.MX8MP)
IMX8MMRM Rev. 3 (i.MX8MM)

Upstream-Status: Pending

Signed-off-by: Stefan Kerkmann <s.kerkmann@pengutronix.de>
---
 include/mach/imx/ocotp-fusemap.h | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/include/mach/imx/ocotp-fusemap.h b/include/mach/imx/ocotp-fusemap.h
index 37f1ee8298c2..3c961998b0fc 100644
--- a/include/mach/imx/ocotp-fusemap.h
+++ b/include/mach/imx/ocotp-fusemap.h
@@ -12,6 +12,22 @@
 #define OCOTP_MAC_ADDR_LOCK		(OCOTP_WORD(0x400) | OCOTP_BIT(8) | OCOTP_WIDTH(2))
 #define OCOTP_GP1_LOCK			(OCOTP_WORD(0x400) | OCOTP_BIT(10) | OCOTP_WIDTH(2))
 #define OCOTP_GP2_LOCK			(OCOTP_WORD(0x400) | OCOTP_BIT(12) | OCOTP_WIDTH(2))
+
+/* available on i.MX8 */
+#define MX8M_OCOTP_GP1_LOCK		(OCOTP_WORD(0x400) | OCOTP_BIT(20) | OCOTP_WIDTH(2))
+#define MX8M_OCOTP_GP2_LOCK		(OCOTP_WORD(0x400) | OCOTP_BIT(22) | OCOTP_WIDTH(2))
+#define MX8M_OCOTP_GP5_LOCK		(OCOTP_WORD(0x400) | OCOTP_BIT(24) | OCOTP_WIDTH(2))
+
+/* available on i.MX8MP */
+#define MX8MP_OCOTP_GP1_LOCK	MX8M_OCOTP_GP1_LOCK
+#define MX8MP_OCOTP_GP2_LOCK	MX8M_OCOTP_GP2_LOCK
+#define MX8MP_OCOTP_GP4_LOCK	(OCOTP_WORD(0x410) | OCOTP_BIT(4)  | OCOTP_WIDTH(2))
+#define MX8MP_OCOTP_GP5_LOCK	(OCOTP_WORD(0x410) | OCOTP_BIT(8)  | OCOTP_WIDTH(2))
+#define MX8MP_OCOTP_GP6_LOCK	(OCOTP_WORD(0x410) | OCOTP_BIT(10) | OCOTP_WIDTH(2))
+#define MX8MP_OCOTP_GP7_LOCK	(OCOTP_WORD(0x410) | OCOTP_BIT(12) | OCOTP_WIDTH(2))
+#define MX8MP_OCOTP_GP8_LOCK	(OCOTP_WORD(0x410) | OCOTP_BIT(14) | OCOTP_WIDTH(2))
+#define MX8MP_OCOTP_GP9_LOCK	(OCOTP_WORD(0x410) | OCOTP_BIT(16) | OCOTP_WIDTH(2))
+
 #define OCOTP_SRK_LOCK			(OCOTP_WORD(0x400) | OCOTP_BIT(14) | OCOTP_WIDTH(1))
 #define OCOTP_ANALOG_LOCK		(OCOTP_WORD(0x400) | OCOTP_BIT(18) | OCOTP_WIDTH(2))
 #define OCOTP_MISC_CONF_LOCK		(OCOTP_WORD(0x400) | OCOTP_BIT(22) | OCOTP_WIDTH(1))
